polars_st/geodataframe.py

```diff
@@ -340,11 +340,12 @@ def to_geopandas(
         import geopandas as gpd
 
         return gpd.GeoDataFrame(
-            self._df.select(geom(geometry_name).st.to_shapely()).to_pandas(
+            self._df.with_columns(geom(geometry_name).st.to_shapely()).to_pandas(
                 use_pyarrow_extension_array=use_pyarrow_extension_array,
-                geometry=geometry_name,
                 **kwargs,
             ),
+            geometry=geometry_name,
+            crs=self._df.select(geom(geometry_name).st.srid())[0, 0],
```

Comment:
This will fail if the DataFrame is empty, and is also wrong in cases where the geometry column contains mixed SRIDs. I'd rather leave that out of this PR, and open a different issue.